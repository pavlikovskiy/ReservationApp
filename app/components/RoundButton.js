/**
 * @flow
 */

import React, { Component } from "react";
import { Button, TouchableOpacity, Text, View, StyleSheet } from "react-native";
import Ionicons from "react-native-vector-icons/Ionicons";

export default class RoundButton extends Component<RoundButtonProps> {
  render() {
    const { onPress, iconSize } = this.props;
    return (
      <View style={styles.controlButtonContainer}>
        <TouchableOpacity style={styles.roundButton} onPress={onPress}>
          <Ionicons
            name={"ios-add"}
            size={iconSize}
            style={{ color: "#3478f6" }}
          />
        </TouchableOpacity>
      </View>
    );
  }
}

type RoundButtonProps = {
  onPress: Function,
  iconSize: number
};

const styles = StyleSheet.create({
  controlButtonContainer: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  },

  roundButton: {
    borderWidth: 1,
    borderColor: "#808080",
    alignItems: "center",
    justifyContent: "center",
    width: 75,
    height: 75,
    backgroundColor: "rgb(235,235,235)",
    borderRadius: 75
  }
});
