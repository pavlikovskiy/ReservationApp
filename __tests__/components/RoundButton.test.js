import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import RoundButton from "../../app/components/RoundButton";
import Ionicons from "react-native-vector-icons/Ionicons";
import { TouchableOpacity } from "react-native";

describe("RoundButton", () => {
  it("should render without issues", () => {
    const component = shallow(<RoundButton onPress={() => {}} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render touchable ", () => {
    const component = shallow(<RoundButton onPress={() => {}} />);

    expect(component.find(TouchableOpacity)).toBeTruthy();
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render with vector icon ", () => {
    const component = shallow(<RoundButton onPress={() => {}} />);

    expect(component.find(Ionicons)).toBeTruthy();
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render with vector icon specific size ", () => {
    const component = shallow(<RoundButton onPress={() => {}} iconSize={70} />);

    expect(component.find(Ionicons)).toBeTruthy();
    expect(component.find(Ionicons).props().size).toBe(70);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
