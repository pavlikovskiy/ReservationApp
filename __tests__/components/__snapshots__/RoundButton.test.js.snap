// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton should render touchable  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(235,235,235)",
        "borderColor": "#808080",
        "borderRadius": 75,
        "borderWidth": 1,
        "height": 75,
        "justifyContent": "center",
        "width": 75,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="ios-add"
      size={12}
      style={
        Object {
          "color": "#3478f6",
        }
      }
    />
  </TouchableOpacity>
</View>
`;

exports[`RoundButton should render with vector icon  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(235,235,235)",
        "borderColor": "#808080",
        "borderRadius": 75,
        "borderWidth": 1,
        "height": 75,
        "justifyContent": "center",
        "width": 75,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="ios-add"
      size={12}
      style={
        Object {
          "color": "#3478f6",
        }
      }
    />
  </TouchableOpacity>
</View>
`;

exports[`RoundButton should render with vector icon specific size  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(235,235,235)",
        "borderColor": "#808080",
        "borderRadius": 75,
        "borderWidth": 1,
        "height": 75,
        "justifyContent": "center",
        "width": 75,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="ios-add"
      size={70}
      style={
        Object {
          "color": "#3478f6",
        }
      }
    />
  </TouchableOpacity>
</View>
`;

exports[`RoundButton should render without issues 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(235,235,235)",
        "borderColor": "#808080",
        "borderRadius": 75,
        "borderWidth": 1,
        "height": 75,
        "justifyContent": "center",
        "width": 75,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="ios-add"
      size={12}
      style={
        Object {
          "color": "#3478f6",
        }
      }
    />
  </TouchableOpacity>
</View>
`;
